stl/inc/cmath

```diff
@@ -953,261 +952,373 @@ _NODISCARD _CRT_SATELLITE_2 float __stdcall __std_smf_hypot3f(float, float, floa
 }
 
-_NODISCARD inline long double riemann_zetal(const long double _Arg) {
+_NODISCARD inline long double riemann_zetal(const long double _Arg) noexcept /* strengthened */ {
     return __std_smf_riemann_zeta(static_cast<double>(_Arg));
 }
 
-_NODISCARD inline double sph_bessel(const unsigned int _Arg1, const double _Arg2) {
+_NODISCARD inline double sph_bessel(const unsigned int _Arg1, const double _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_bessel(_Arg1, _Arg2);
 }
 
-_NODISCARD inline float sph_besself(const unsigned int _Arg1, const float _Arg2) {
+_NODISCARD inline float sph_besself(const unsigned int _Arg1, const float _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_besself(_Arg1, _Arg2);
 }
 
-_NODISCARD inline long double sph_bessell(const unsigned int _Arg1, const long double _Arg2) {
+_NODISCARD inline long double sph_bessell(const unsigned int _Arg1, const long double _Arg2) noexcept
+/* strengthened */ {
     return __std_smf_sph_bessel(_Arg1, static_cast<double>(_Arg2));
 }
 
-_NODISCARD inline double sph_legendre(const unsigned int _Arg1, const unsigned int _Arg2, const double _Theta) {
+_NODISCARD inline double sph_legendre(const unsigned int _Arg1, const unsigned int _Arg2, const double _Theta) noexcept
+/* strengthened */ {
     return __std_smf_sph_legendre(_Arg1, _Arg2, _Theta);
 }
 
-_NODISCARD inline float sph_legendref(const unsigned int _Arg1, const unsigned int _Arg2, const float _Theta) {
+_NODISCARD inline float sph_legendref(const unsigned int _Arg1, const unsigned int _Arg2, const float _Theta) noexcept
+/* strengthened */ {
     return __std_smf_sph_legendref(_Arg1, _Arg2, _Theta);
 }
 
 _NODISCARD inline long double sph_legendrel(
-    const unsigned int _Arg1, const unsigned int _Arg2, const long double _Theta) {
+    const unsigned int _Arg1, const unsigned int _Arg2, const long double _Theta) noexcept /* strengthened */ {
     return __std_smf_sph_legendre(_Arg1, _Arg2, static_cast<double>(_Theta));
 }
 
-_NODISCARD inline double sph_neumann(const unsigned int _Arg1, const double _Arg2) {
+_NODISCARD inline double sph_neumann(const unsigned int _Arg1, const double _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_neumann(_Arg1, _Arg2);
 }
 
-_NODISCARD inline float sph_neumannf(const unsigned int _Arg1, const float _Arg2) {
+_NODISCARD inline float sph_neumannf(const unsigned int _Arg1, const float _Arg2) noexcept /* strengthened */ {
     return __std_smf_sph_neumannf(_Arg1, _Arg2);
 }
 
-_NODISCARD inline long double sph_neumannl(const unsigned int _Arg1, const long double _Arg2) {
+_NODISCARD inline long double sph_neumannl(const unsigned int _Arg1, const long double _Arg2) noexcept
+/* strengthened */ {
     return __std_smf_sph_neumann(_Arg1, static_cast<double>(_Arg2));
 }
+_STD_END
 
+#define _GENERIC_MATH_SPECIAL1(NAME)                                                  \
+    _STD_BEGIN                                                                        \

```

Comment:
Why bracket the macro definitions with `_STD_BEGIN` and `_STD_END` instead of simply leaving the namespace open around the uses of the macros?

Reply:
I'm a bit afraid of `#define`ing macros inside a namespace...
Changing to expand these macros within `_STD_BEGIN` and `_STD_END`, but still leaving `#define` and `#undef` outside of a namespace definition.